freva-rest/src/freva_rest/databrowser_api/endpoints.py

```diff
@@ -511,3 +505,158 @@ async def delete_user_data(
+        auth.create_auth_dependency(required=False), scopes=["oidc.claims"]
+    ),
+
+) -> FlavourListResponse:
+    """Get available flavours for the current user.
+
+    This endpoint retrieves all flavours that are accessible to the current user,
+    including built-in flavours (freva, cmip5, cmip6, cordex, user) and any custom
+    flavours created by the user or made globally available by administrators.
+
+    The results can be optionally filtered by flavour name to get a specific
+    flavour, or by owner to show only global flavours or user-specific ones.
+    For unauthenticated users, only global flavours are returned.
+    """
+    solr = Solr.validate_flavour_parameters(server_config, request)
+    user_name = (current_user.preferred_username
+                 if current_user and current_user.preferred_username
+                 else "global")
+
+    all_flavours = await solr.get_all_flavours(user_name, flavour_name, owner)
+
+    return FlavourListResponse(
+        total=len(all_flavours),
+        flavours=all_flavours
+    )
+
+
+@app.delete(
+    "/api/freva-nextgen/databrowser/flavours/{flavour_name}",
+    tags=["Data search"],
+    status_code=200,
+    response_model=FlavourDeleteResponse,
+    responses={
+        401: {"description": "Unauthorised / not a valid token."},
+        422: {"description": "Invalid flavour name."},
+        403: {"description": "Forbidden - cannot delete global flavours."},
+        404: {"description": "Flavour not found."},
+        500: {"description": "Internal server error - failed to delete flavour."},
+    },
+)
+async def delete_custom_flavour(
+    flavour_name: str = Path(
+        ..., description="Name of the flavour to delete",
+        examples=["nextgem", "custom_project"]
+    ),
+    current_user: TokenPayload = Security(
+        auth.create_auth_dependency(), scopes=["oidc.claims"]
+    ),
+) -> FlavourDeleteResponse:
+    """Delete a custom flavour definition.
+
+    This endpoint allows authenticated users to delete custom flavour definitions
+    that they have previously created. Regular users can only delete their own
+    personal flavours, while administrators can delete both personal and global
+    flavours.
+
+    Built-in flavours (freva, cmip5, cmip6, cordex, user) cannot be deleted as
+    they are system-defined and always available. Once a custom flavour is
+    deleted, it will no longer be available for use in databrowser searches
+    and will be removed from the flavour listings.
+    """
+    if flavour_name.lower() in [f.lower() for f in BUILTIN_FLAVOURS]:
+        logger.error(
+            "Attempt to delete built-in flavour: %s by user: %s",
+            flavour_name,
+            current_user.preferred_username
+        )
+        raise HTTPException(
+            status_code=422,
+            detail=f"Cannot delete built-in flavour '{flavour_name}'"
+        )
+
+    is_global = server_config.is_admin_user(current_user)
```

Comment:
This assumes that admins can only add global falvours. Why?

Reply:
There is a grammatically nuance in my comment. I said:

Admins can *only* delete global flavours.

I guess you understood:

*Only* admins can delete global flavours.

I am not questioning at all what you said. But what I said was the following: When I am admin my flavour will always become global. This logic could be ok but then you'd have to also implement this logic in the flavour add functionality. Because the add functionality checks if you're admin *and* your flavour should become global. Sot there is a chance for an admin to add a private flavour. Here we only check if you're admin and assume if we are then it's global. So there is a clash in the code.